packages/vite/src/node/config.ts

```diff
@@ -265,6 +276,11 @@ export async function resolveConfig(
     }
   }
 
+  // update known js src extensions
+  config.knownJsSrcExtensions?.forEach((ext) =>
+    KNOWN_JS_SRC_EXTENSIONS.add(ext.startsWith('.') ? ext : `.${ext}`)
+  )
```

Comment:
I think we shouldn't modify `KNOW_JS_SRC_EXTENTIONS` here, but create a new `knownJsSrcExtensions` entry in ResolvedConfig that have both the default plus the added by plugins. You should be able to run Vite build programmatically with different plugins without interfering with each other (I have doubts about the cache, but for the rest, it should be independent). This would imply that we have to pass this to `isJsRequest`, I hope there isn't that much wiring to do.

Reply:
that wiring was the reason i resorted to this. but you're right that modifying that isn't stable esp. if config changes. Should it be a string array in the ResolvedConfig or a Set<String> or should ResolvedConfig have a `isJSRequest` function (like assetsInclude already is)